packages/commerce/src/types/checkout.ts

```diff
@@ -1,57 +1,97 @@
 import type { UseSubmitCheckout } from '../checkout/use-submit-checkout'
 import type { Address, AddressFields } from './customer/address'
 import type { Card, CardFields } from './customer/card'
+import type { LineItem } from './cart'
 
-// Index
-export type Checkout = any
+export type Checkout = {
+  /**
+   * Indicates if the payment has been submitted.
+   */
+  hasPayment: boolean
+  /**
+   * Indicates if the checkout has shipping information collected.
+   */
+  hasShipping: boolean
+  /**
+   * The unique identifier for the address that the customer has selected for shipping.
+   */
+  addressId: string
+  /**
+   * The list of payments that the customer has selected for the checkout.
+   */
+  payments?: Card[]
+  /**
+   * The unique identifier of the card that the customer has selected for payment.
+   */
+  cardId?: string
+  /**
+   * List of items in the checkout.
+   */
+  lineItems?: LineItem[]
+}
+
+export type CheckoutBody = {
+  /**
+   * The unique identifier for the cart.
+   */
+  cartId?: string
+  /**
+   * The Card information.
+   * @see CardFields
+   */
+  card: CardFields
+  /**
+   * The Address information.
+   * @see AddressFields
+   */
+  address: AddressFields
+}
 
-export type CheckoutTypes = {
+export interface CheckoutTypes {
   card?: Card | CardFields
   address?: Address | AddressFields
   checkout?: Checkout
   hasPayment?: boolean
   hasShipping?: boolean
 }
```

Comment:
What is `CheckoutTypes` needed for?

Reply:
I forgot to delete this, it was used before at the generics.

`export type SubmitCheckoutHook<T extends CheckoutTypes = CheckoutTypes> = {`